apis/v1beta1/config_test.go

```diff
@@ -26,11 +30,227 @@ import (
 	"github.com/stretchr/testify/require"
 	go_yaml "gopkg.in/yaml.v3"
 	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/kubernetes/scheme"
 	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
 	"sigs.k8s.io/yaml"
 )
 
+func TestCreateConfigInKubernetesEmptyValues(t *testing.T) {
+	testScheme := scheme.Scheme
+	err := AddToScheme(testScheme)
+	require.NoError(t, err)
+
+	testEnv := &envtest.Environment{
+		CRDDirectoryPaths: []string{filepath.Join("..", "..", "config", "crd", "bases")},
+	}
+
+	cfg, err := testEnv.Start()
+	require.NoError(t, err)
+	defer func() {
+		errStop := testEnv.Stop()
+		require.NoError(t, errStop)
+	}()
+
+	k8sClient, err := client.New(cfg, client.Options{Scheme: testScheme})
+	if err != nil {
+		fmt.Printf("failed to setup a Kubernetes client: %v", err)
+		os.Exit(1)
```

Comment:
ive found [this fake client package](https://github.com/open-telemetry/opentelemetry-operator/blob/7ee69c59d6727bddf8b1af71da544eac7df08a9b/cmd/operator-opamp-bridge/operator/client_test.go#L46-L59) a lot more usable